### ISSUE TITLE
Use-case: Metabase as a report engine

### ISSUE THREAD

**Issue by vi-vsn** [metabase/metabase#8460]
Hello,

Just wanted to bring to your attention a use-case that we have for Metabase, so that you are aware that some people use it for this, along with a couple of requests that could make it better.

Basically, while we do have a share of dashboards - we mostly reply on the Metabase as more of a traditional reporting engine, where people are directed to questions that have been constructed for them. Overall it works pretty satisfactory for this purpose with a couple of inconveniences:

- Wide grids that contain many columns - are hard to follow which row you are looking at. It would  be helpful if we could enable alternate row highlighting.

- Some questions are meant to take a relatively long time to run - but they time out. Would it be possible to expose the timeout setting in the admin panel? I don't anticipate setting it that high, but up to 10 minutes would be ideal.

- Only first 10,000 rows are shown. This is reasonable and I can often work around that, but ideally there shouldn't be a limit - it would be ideal if tables of arbitrary size can be viewed, perhaps with dynamic paging.

- CSV exports of large grids can frequently time out. It would be ideal if this was solved somehow so grids of arbitrary size can be exported and downloaded.

- Being able to attractively print large multi-page reports would be pretty helpful too. Currently the workflow involves exporting to Excel and printing from there.

- I have also mentioned this earlier (https://github.com/metabase/metabase/issues/7089) but in a different context. Being able to at least sort by clicking the the column headers for the Native Question is really a must have and I get the most complaints about that from the users.

- Quite frequently I wish to be able to output "clickable" URL's in my report output, so that the users don't need to copy/paste a url that they see in the grid, but rather simply click on it.

Thank you!

**Comment by vi-vsn** [comment]
One thing I wanted to add, is that even though Metabase was perhaps not necessarily intended for this kind of use, I found that it was pretty much the only tool out there, proprietary or open, that could deliver the ease of use and convenience that I was looking for. When i started evaluating solutions I was looking for a Linux-based alternative for Microsoft SQL Server Reporting Services and discovered to my dismay that everything I have tried was really disappointing (have you seen the state of BiRT and it's spin-offs?). I am posting this here, so that perhaps you see an opportunity to expand into an area that competition does not adequately cover. Thanks for your hard work.

**Comment by vi-vsn** [comment]
I hope it's ok to bump an old post, since it's been a year and a lot has changed in Metabase....

My number one request - is giving end users ability to sort and filter results that came from SQL Query-based question (without having to modify the underlying SQL).

**Comment by flamber** [comment]
Hi @vi-vsn 
While I agree with some of the things, it's difficult to handle issues with multiple points.
So let's break it out, so it's more manageable.
1. Alternate row highlighting - seems like a good idea. Could you create a separate feature request?
2. There shouldn't be any timeouts. What is causing the timeout? Database, reverse-proxy, Metabase, ...? Check the browser console or Metabase log for additional context.
3. Row limit. Sounds like a mix of #4564 and #5504.
4. CSV timeout. Most likely the same root problem as number 2.
5. Print multi-page reports. I'm not quite sure what you're looking for. It sounds like #3770.
6. Native column sort. You can get most of the way with Saved Questions (with it's own limitations, like the one you link to). But making Metabase understand what to replace in an advanced query would not be easy - each driver would need take care of how that's handled for each of their respected languages. If that's your major issue among all of these points, then you should create a separate feature request for that.
7. Clickable links. Since version 0.30 (I think) column formatting was added, so the entire column could be defined as a clickable link. If you're looking for being able having parts of a column text as link, then that sounds like #7188.

**Comment by flamber** [comment]
Closing this, since it's difficult to manage multiple items per issue. I've linked to several other issues.
